server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesHotThreadsAction.java

```diff
@@ -23,16 +24,59 @@
 
 import java.io.IOException;
 import java.util.List;
+import java.util.Locale;
 
 import static org.elasticsearch.rest.RestRequest.Method.GET;
 
 public class RestNodesHotThreadsAction extends BaseRestHandler {
 
+    private static final String formatDeprecatedMessageWithoutNodeID = "[%s] is a deprecated endpoint. " +
```

Comment:
just a thought. we could just use one message for all. We have a used endpoint as a key, `API_GET_/_nodes/{nodeid}/hot_threads` 
not as readable as here, but might save some effort
but it looks good now too!

Reply:
I basically did a revert + edit of the commit that removed the apis, so it was cheap to do it as is (and keeps the code mostly in line with the way it is on 7.x).